docs/src/design/ten-thousand-foot-view.md

```diff
@@ -0,0 +1,211 @@
+
+*To complete initialization of a map component, we need to initialize a
+map, style with source and layers, and observers.*
+
+### Layer
+
+*Layer* is an overloaded terminology in the MapLibre GL Native's
+context. *Layer* can mean any of the following:
+
+1.  From the point of view of data that needs to be rendered on map,
+    each map tile is composed of *layers* of data. Each layer, in this
+    context, contains *features*[^4] to be rendered in a map. These
+    features are defined by a *source.* Each *layer* is tied to a
+    *source*.
+
+2.  From the point of view of *style*, a style's layer property list all
+    the *layers* available in that *style*. A single style sheet can be
+    applied atop one or many layers. This layer definition converges
+    data to be rendered and the style to be applied on said layer.
+
+When this document uses the word layer in context of rendering, it
+refers to the definition mentioned in 2.
+
+### Glyphs
+
+A *glyph* is a single representation of a character. A font is a map of
+characters[^5]. Map tiles use labels of text to show name of cities,
+administrative boundaries, or street names. Map tiles also need to show
+icons for amenities like bus stops and parks. A map style uses character
+map from fonts to display labels and icons. Collectively these are
+called *glyphs.* A glyph is loaded as a set of signed-distance-fields in
+protobuf format. For curious readers, a signed-distance-field which
+takes a position and desired glyph as input and returns the distance
+from that position to the desired glyph. This is used in a rendering
+technique called atlasing. To build a glyph atlas, MapLibre GL Native
+builds a single image that will be uploaded as a texture to the GPU
+along with the rest of the data for the tile.
+
+### Actor Framework
+
+MapLibre GL Native is used in mobile platforms. To be performant in
+underpowered environments, MapLibre GL Native tries to leverage message
+passing across threads to render frames asynchronously. The threading
+architecture in implementation realizes this using the Actor
+interface[^6]. In reality the messages are raw pointers. This ranges
+from raw message to actionable message. By actionable message this
+document means *anonymous functions* that are passed as messages between
+actors. These arbitrary messages are *immutable* by design. The current
+implementation of the Actor framework is done through two major
+components, a *MailBox* and an *Actor*. A *MailBox* is attached to a
+specific type of *Message*. In the current implementation, these are
+rendering events that render layers, sources, and subsequently tiles. An
+*Actor* is an independent thread passing messages to others.
+
+### Renderer
+
+A *Map Renderer* translates geospatial features in a vector or raster
+tile to rendered or rasterized map tiles shown in a *slippy map.*
+MapLibre GL Native uses a *Renderer* component to translate map tiles
+fetched from a tile server to a rendered map.
```

Comment:
Is it strictly true that it results in `map tiles`? Wouldn't it be true to say _conceptually_ the rendered map has no concept of a tile (other than when you set it in tile mode) rather that is has a viewport on a map (typically composed on a size + center + other camera settings)?

Reply:
Yes, the finalized viewport does not have a concept of tiles. I went back and forth with this. Since this is the early text for a reader, I did not dive too deep into rendering implmentation. The code processes one tile at a time and renders the tile in the tile coordinate to produe a single viewport.

But yes, you are right. I do talk about this going forward in the document. :)

Reply:
I am willing to amend the text. I will think more about it. :) Focusing on your SDF comment. I think that one is a bigger crime from my part. 🙈